docs/source/developer/debugging.rst

```diff
@@ -0,0 +1,130 @@
+happening in Python, but this depends on examining the backtrace closely. For
+example, in a backtrace corresponding to the above error, we see items in the
+backtrace such as:
+
+.. code-block::
+
+   #18 0x00000000002722da in slot_tp_call (
+       self=<_wrap_impl(_callable=<_wrap_missing_loc(func=<function at remote
+       0x1cf66c20>) at remote 0x1d200bd0>, _imp=<function at remote 0x1d0e7440>,
+       _context=<CUDATargetContext(address_size=64,
+       typing_context=<CUDATypingContext(_registries={<Registry(functions=[<type
+       at remote 0x65be5e0>, <type at remote 0x65be9d0>, <type at remote
+       0x65bedc0>, <type at remote 0x65bf1b0>, <type at remote 0x8b78000>, <type
+       at remote 0x8b783f0>, <type at remote 0x8b787e0>, <type at remote
+       0x8b78bd0>, <type at remote 0x8b78fc0>, <type at remote 0x8b793b0>, <type
+       at remote 0x8b797a0>, <type at remote 0x8b79b90>, <type at remote
+       0x8b79f80>, <type at remote 0x8b7a370>, <type at remote 0x8b7a760>, <type
+       at remote 0x8b7ab50>, <type at remote 0x8b7af40>, <type at remote
+       0x8b7b330>, <type at remote 0x8b7b720>, <type at remote 0x8b7bf00>, <type
+       at remote 0x8b7c2f0>, <type at remote 0x8b7c6e0>], attributes=[<type at
+       remote 0x8b7cad0>, <type at remote 0x8b7cec0>, <type at remote
+       0x8b7d2b0>, <type at remote 0x8b7d6a0>, <type at remote 0x8b7da90>,
+       <t...(truncated),
+       args=(<Builder(_block=<Block(parent=<Function(parent=<Module(context=<Context(scope=<NameScope(_useset={''},
+       _basenamemap={}) at remote 0xbb5ae10>, identified_types={}) at remote
+       0xbb5add0>, name='cuconstRecAlign$7',
+       data_layout='e-p:64:64:64-i1:8:8-i8:8:8-i16:16:16-i32:32:32-i64:64:64-f32:32:32-f64:64:64-v16:16:16-v32:32:32-v64:64:64-v128:128:128-n16:32:64',
+       scope=<NameScope(_useset={'',
+       '_ZN08NumbaEnv5numba4cuda5tests6cudapy13test_constmem19cuconstRecAlign$247E5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE',
+       '_ZN5numba4cuda5tests6cudapy13test_constmem19cuconstRecAlign$247E5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE'},
+       _basenamemap={}) at remote 0x1d27bf10>, triple='nvptx64-nvidia-cuda',
+       globals={'_ZN08NumbaEnv5numba4cuda5tests6cudapy13test_constmem19cuconstRecAlign$247E5ArrayIdLi1E1C7mutable7ali...(truncated),
+       kwds=0x0)
+
+We can see some of the arguments, in particular the names of the compiled functions, e.g::
+
+   _ZN5numba4cuda5tests6cudapy13test_constmem19cuconstRecAlign$247E5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE5ArrayIdLi1E1C7mutable7alignedE
+
+We can run this through ``c++filt`` to see a more human-readable representation::
```

Comment:
` set print asm-demangle on` ?

Reply:
This doesn't do anything for me, perhaps because those strings are part of char strings in function arguments (I think it would be surprising for GDB to filter strings in memory before printing them).